src/compose/app.ts

```diff
@@ -774,16 +778,36 @@ class AppImpl implements App {
 				return [];
 			}
 		} else if (target.config.running !== current.config.running) {
-			// Take lock for all services before starting/stopping container
-			if (!servicesLocked) {
-				this.services.concat(targetServices).forEach((s) => {
-					appsToLock[target.appId].add(s.serviceName);
-				});
-				return [];
-			}
 			if (target.config.running) {
+				// if the container has a reboot
+				// required label and the boot time is before the creation time, then
+				// return a 'noop' to ensure a reboot happens before starting the container
+				const requiresReboot =
+					target.config.labels?.['io.balena.update.requires-reboot'] !== null &&
```

Comment:
```suggestion
					target.config.labels?.['io.balena.update.requires-reboot'] === 'true' &&
```
Should we be more restrictive with the value of the label? Or at least, not return true for `undefined`:
```suggestion
					target.config.labels?.['io.balena.update.requires-reboot'] != null &&
```

Reply:
You are right, we use `checkTruthy` for the other lablels so it makes sense to be consistent

https://github.com/balena-os/balena-supervisor/blob/81b307510d758c9390836acd2d8547e1263eaf35/src/compose/utils.ts#L413